catalog/schematron/catalog-schematron-plugin/src/main/java/ddf/services/schematron/SchematronValidationService.java

```diff
@@ -118,54 +110,44 @@ private void updateValidators() throws SchematronInitializationException {
         }
     }
 
-    private Templates compileSchematronRules(String schematronFileName)
-            throws SchematronInitializationException {
+    private Templates compileSchematronRules(String schematronFileName) throws SchematronInitializationException {
 
         Templates template;
         File schematronFile = new File(schematronFileName);
         if (!schematronFile.exists()) {
-            throw new SchematronInitializationException(
-                    "Could not locate schematron file" + schematronFileName);
+            throw new SchematronInitializationException("Could not locate schematron file" + schematronFileName);
         }
 
         try {
-            URL schUrl = schematronFile.toURI()
-                    .toURL();
+            URL schUrl = schematronFile.toURI().toURL();
             Source schSource = new StreamSource(schUrl.toString());
 
             // Stage 1: Perform inclusion expansion on Schematron schema file
-            DOMResult stage1Result = performStage(schSource,
-                    getClass().getClassLoader()
-                            .getResource("iso-schematron/iso_dsdl_include.xsl"));
+            DOMResult stage1Result = performStage(schSource, getClass().getClassLoader().getResource("iso-schematron/iso_dsdl_include.xsl"));
             DOMSource stage1Output = new DOMSource(stage1Result.getNode());
 
             // Stage 2: Perform abstract expansion on output file from Stage 1
-            DOMResult stage2Result = performStage(stage1Output,
-                    getClass().getClassLoader()
-                            .getResource("iso-schematron/iso_abstract_expand.xsl"));
+            DOMResult stage2Result = performStage(stage1Output, getClass().getClassLoader().getResource("iso-schematron/iso_abstract_expand.xsl"));
             DOMSource stage2Output = new DOMSource(stage2Result.getNode());
 
             // Stage 3: Compile the .sch rules that have been prepocessed by Stages 1 and 2 (i.e.,
             // the output of Stage 2)
-            DOMResult stage3Result = performStage(stage2Output,
-                    getClass().getClassLoader()
-                            .getResource("iso-schematron/iso_svrl_for_xslt2.xsl"));
+            DOMResult stage3Result = performStage(stage2Output, getClass().getClassLoader().getResource("iso-schematron/iso_svrl_for_xslt2.xsl"));
             DOMSource stage3Output = new DOMSource(stage3Result.getNode());
 
+            // Setting the system ID let's us resolve relative paths in the schematron files.
+            stage3Output.setSystemId(schematronFileName);
```

Comment:
In case anyone else has as much trouble finding it as me, I believe this is *the* salient difference in this pull request, correct? Looks like everything else is formatting/tests.